presto-jdbc/src/main/java/com/facebook/presto/jdbc/PrestoConnection.java

```diff
@@ -553,14 +557,19 @@ public void abort(Executor executor)
     public void setNetworkTimeout(Executor executor, int milliseconds)
             throws SQLException
     {
-        throw new SQLFeatureNotSupportedException("setNetworkTimeout");
+        checkOpen();
+        if (milliseconds < 0) {
+            throw new SQLException("Timeout is negative");
+        }
+        networkTimeoutMillis.set(milliseconds);
```

Comment:
Small caveat. The documentation says: 

> However if the network timeout expires [...] the connection will be marked as closed, any resources held by the connection will be released and both the connection and statement will be unusable.

This part of the API is not implemented. Maybe mention that in the method-level javadoc?

Reply:
Interesting, I missed that part. However, I think the current implementation. We don't hold on to resources. The part about connection objects being invalid is seems to be about expectations for users rather than prescriptive for driver implementors.

That said, if you think we need to implement this, it's not hard to do.